dashboard/app/models/user.rb

```diff
@@ -926,15 +948,47 @@ def self.send_reset_password_instructions(attributes={})
       return user
     end
 
-    user = find_by_email_or_hashed_email(attributes[:email]) || User.new(email: attributes[:email])
-    if user && user.persisted?
```

Comment:
That's a good question, though IIRC it's deliberate (sorry, this is an older set of changes that I've resurrected) - notice that above, User.new(...) is called if the email lookup fails, that's the case where the user would not be persisted that this check guards against, which is replaced by the not_found_user path below.